### PR TITLE
(feat): Adding Crd validation for components.runner ( minProperties =2)

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -333,6 +333,8 @@ func newRunnerPodForCR(ce chaosTypes.EngineInfo) (*corev1.Pod, error) {
 	//Initiate the Engine Info, with the type of runner to be used
 	if ce.Instance.Spec.Components.Runner.Type == "ansible" {
 		return newAnsibleRunnerPodForCR(ce)
+	} else if ce.Instance.Spec.Components.Runner.Type == "go" {
+		return newAnsibleRunnerPodForCR(ce)
```
Comment (ksatchit) [inline pkg/controller/chaosengine/chaosengine_controller.go]: you mean go here?

**Comment by ksatchit** [comment]
Was more of a CR level validation, but this change is harmless too.

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 2]
```diff
@@ -333,6 +333,8 @@ func newRunnerPodForCR(ce chaosTypes.EngineInfo) (*corev1.Pod, error) {
 	//Initiate the Engine Info, with the type of runner to be used
 	if ce.Instance.Spec.Components.Runner.Type == "ansible" {
 		return newAnsibleRunnerPodForCR(ce)
+	} else if ce.Instance.Spec.Components.Runner.Type == "go" {
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Why this check is required? As we already have the check for ansible

**Comment by rahulchheda** [comment]
Not needed, specifying "go" without this PR, will also spawn `chaos-runner`

File: pkg/controller/chaosengine/chaosengine_controller.go (hunk 2 above)
Comment (shubhamchaudhary) [reply pkg/controller/chaosengine/chaosengine_controller.go]: It is written to make sure that If anyone will look at the code, they can understand that it is only supporting go & ansible runners.
cc: @ksatchit

File: deploy/chaos_crds.yaml [hunk 3]
```diff
@@ -69,6 +69,7 @@ spec:
                       type: string
                 runner:
                   type: object
+                  minProperties: 2
```
Comment (ksatchit) [inline deploy/chaos_crds.yaml]: @shubhamchaudhary name of keys (type & image)

**Comment by ksatchit** [comment]
Additional (combinatorial) validation will be handled by the validation admission controller.

**Pull request closed without merging**